lib/CL/devices/basic/basic.c

```diff
@@ -719,14 +719,21 @@ pocl_basic_fill_rect (void *data,
                 + buffer_slice_pitch * k, fill_pixel, pixel_size);
 }
 
+typedef unsigned long v2ul __attribute__ ((vector_size (16)));
```

Comment:
If you did not measure any major speedups on any platform, the original loops version might overall ne better. The biggest improvement from this comes from the fact that you communicate to the compiler that the data is aligned (which it otherwise cannot tell). Anyways, proper vectorizers will specialize or add a prologue in that case. Thus, I'd use a clean for loop here instead or a libc function (even better), if it can be implemented with such.